Classes/Service/ContentService.php

```diff
@@ -345,24 +345,38 @@ protected function initializeRecordByNewAndOldAndLanguageUids(
                 // look for the translated version of the parent record indicated
                 // in this new, translated record. Below, we adjust the parent UID
                 // so it has the UID of the translated parent if one exists.
-                $translatedParents = (array) $this->workspacesAwareRecordService->get(
-                    'tt_content',
-                    'uid,sys_language_uid',
-                    "t3_origuid = '" . $oldRecord['tx_flux_parent'] . "'" . BackendUtility::deleteClause('tt_content')
-                );
-                foreach ($translatedParents as $translatedParent) {
-                    if ($translatedParent['sys_language_uid'] == $newLanguageUid) {
-                        // set $translatedParent to the right language ($newLanguageUid):
-                        break;
+                unset($copiedParent);
+                if (0 < (integer) $oldRecord['tx_flux_parent']) {
+                    // Search for a container element which has been copied from
+                    // the container referred to by the original element
+                    // (oldRecord) and exists in the target language.
+                    // Note that this cannot guarantee the correct container
+                    // to be found and used (using copies for translations
+                    // leads to ambiguous results by principle).
+                    // This is the old method which used to implement a fix for
+                    // #816 and #796.
+                    $copiedParents = (array) $this->workspacesAwareRecordService->get(
+                        'tt_content',
+                        'uid,sys_language_uid',
+                        "t3_origuid = '" . (integer) $oldRecord['tx_flux_parent'] . "' AND " .
+                            "pid = '" . (integer) $row['pid'] . "' AND " .
+                            "CType = 'fluidcontent_content' AND " .
```

Comment:
Hm, maybe I'm confusing things but... While the content that is being translated may be of any `CType`, this is about searching for the container to put it into. Since that container's `uid` needs to go into `tx_flux_parent`. I thought the only `$copiedParents` that make any sense are `fluidcontent` (= FCEs, hopefully ones that actually have a container column of the matching name `tx_flux_column` which has been copied from the to-be-translated record).

For example, I could translate a record with CType `news` which is inside a container, but a CType `news` would not define flux content columns and thus is an invalid container (i.e. `tx_flux_parent`) record?

You're right about not hard-coding extension-related things - if someone would fork `fluidcontent` and rename it `fluidcontent2`, the above code would fail. Is there any better (and still easy) way to do a basic check if the chosen records are capable of being used as containers or should that extra check simply be removed?